src/Features/Core/Portable/PullMemberUp/MembersPuller.cs

```diff
@@ -0,0 +1,408 @@
+            SolutionEditor solutionEditor,
+            SyntaxNode destinationSyntaxNode,
+            CancellationToken cancellationToken)
+        {
+            // Members and destination may come from different documents,
+            // EditorMap is used to save and group all the editors will be used.
+            var editorMapBuilder = ImmutableDictionary.CreateBuilder<SyntaxTree, DocumentEditor>();
+            // One member may have multiple syntaxNodes (e.g partial method).
+            // SyntaxMap is used to find the syntaxNodes need to be changed more easily.
+            var syntaxMapBuilder = ImmutableDictionary.CreateBuilder<ISymbol, SyntaxNode[]>();
+
+            foreach (var memberAnalysisResult in result.MemberAnalysisResults)
+            {
+                var tasks = memberAnalysisResult.Member.DeclaringSyntaxReferences.SelectAsArray(@ref => @ref.GetSyntaxAsync(cancellationToken));
+                var allSyntaxes = await Task.WhenAll(tasks).ConfigureAwait(false);
+                await AddEditorsToEditorMapBuilderAsync(solution, solutionEditor, editorMapBuilder, allSyntaxes).ConfigureAwait(false);
+                syntaxMapBuilder.Add(memberAnalysisResult.Member, allSyntaxes);
+            }
+
+            await AddEditorsToEditorMapBuilderAsync(
+                solution,
+                solutionEditor,
+                editorMapBuilder,
+                new[] { destinationSyntaxNode }).ConfigureAwait(false);
+            return (editorMapBuilder.ToImmutableDictionary(), syntaxMapBuilder.ToImmutableDictionary());
+        }
+
+        private async Task AddEditorsToEditorMapBuilderAsync(
+            Solution solution,
+            SolutionEditor solutionEditor,
+            ImmutableDictionary<SyntaxTree, DocumentEditor>.Builder mapBuilder,
+            SyntaxNode[] syntaxNodes)
+        {
+            foreach (var syntax in syntaxNodes)
+            {
+                if (!mapBuilder.ContainsKey(syntax.SyntaxTree))
+                {
+                    var editor = await solutionEditor.GetDocumentEditorAsync(solution.GetDocumentId(syntax.SyntaxTree)).ConfigureAwait(false);
+                    mapBuilder.Add(syntax.SyntaxTree, editor);
+                }
+            }
+        }
+
+        /// <summary>
+        ///  This method is used to check whether the selected member overrides the member in destination.
+        ///  It just checks the members directly declared in the destination.
+        /// </summary>
+        private bool IsSelectedMemberDeclarationAlreadyInDestination(ISymbol selectedMember, INamedTypeSymbol destination)
+        {
+            if (destination.TypeKind == TypeKind.Interface)
+            {
+                return IsSelectedMemberDeclarationAlreadyInDestinationInterface(selectedMember, destination);
+            }
+            else
+            {
+                return IsSelectedMemberDeclarationAlreadyInDestinationClass(selectedMember, destination);
+            }
```

Comment:
note: you're in a class, but you're cmmonly switching behavior depending on the value of destination.  This could be modeled pretty cleanly as an abstract-base-class, with abstract methods.   i.e. you have the base AbstractPulUpToType class, and PullUpToClass and PullUpToInterface subclasses.  tehse would specialize their methods appropriately.

You do not have to do this.  it's just a thought on how this might be easier to break down.